virttest/libvirt_xml/backup_xml.py

```diff
@@ -0,0 +1,117 @@
+"""
+Module simplifying manipulation of XML described at
+http://libvirt.org/formatbackup.html
+"""
+from virttest import xml_utils
+
+from virttest.libvirt_xml import base, accessors, xcepts
+
+
+class BackupXML(base.LibvirtXMLBase):
+    """
+    Domain backup XML class
+
+    Properties:
+        mode:
+            string (push/pull), backup mode
+        incremental:
+            string, incremental backup based on which checkpoint
+        server:
+            dict, keys: transport, socket, name, port
+    """
+
+    __slots__ = ('mode', 'incremental', 'server', 'disks')
+
+    __uncompareable__ = base.LibvirtXMLBase.__uncompareable__
+
+    __schema_name__ = "domainbackup"
+
+    def __init__(self, virsh_instance=base.virsh):
+        accessors.XMLAttribute('mode', self, parent_xpath='/',
+                               tag_name='domainbackup', attribute='mode')
+        accessors.XMLElementText('incremental', self, parent_xpath='/',
+                                 tag_name='incremental')
+        accessors.XMLElementDict('server', self, parent_xpath='/',
+                                 tag_name='server')
+        super(self.__class__, self).__init__(virsh_instance=virsh_instance)
+        self.xml = '<domainbackup/>'
+
+    def set_disks(self, value_list):
+        """
+        Set the disk sub-elements to the backup xml
+
+        :param value_list: The list of the disk info
+        """
+        for value in value_list:
+            value_type = type(value)
+            if not issubclass(value_type, self.DiskXML):
+                raise xcepts.LibvirtXMLError("Value '%s' Must be a (sub)class of DiskXML,"
+                                             "but not a '%s' type"
+                                             % (str(value), str(value_type)))
+        exist_disks = self.xmltreefile.find('disks')
+        if exist_disks is not None:
+            self.del_disks()
+        if len(value_list) > 0:
+            disks_element = xml_utils.ElementTree.SubElement(
+                    self.xmltreefile.getroot(), 'disks')
+            for disk in value_list:
+                disk_element = disk.xmltreefile.getroot()
+                disks_element.append(disk_element)
+        self.xmltreefile.write()
+
+    def del_disks(self):
+        """
+        Del all disk sub-elements form the backup xml
+        """
+        self.xmltreefile.remove_by_xpath('/disks', remove_all=True)
+        self.xmltreefile.write()
+
+    class DiskXML(base.LibvirtXMLBase):
```

Comment:
Can we reuse Disk under disk.py ?

Reply:
This disk xml is different from the virtual disk xml in vm, it's mainly to indicate if backup of that disk will be executed or not, and indicate some params for backup job, will be as following:
```
<domainbackup>
  <disks>
    <disk name='vda' backup='yes'/>
    <disk name='vdb' type='file'>
      <target file='/path/to/vdb.backup'/>
      <driver type='raw'/>
    </disk>
    <disk name='vdc' backup='no'/>
  </disks>
</domainbackup>
```